src/s3-storage-provider.js

```diff
@@ -13,9 +13,17 @@ let _ = require('lodash');
  * wrapper.  Maybe we should consider adding this wrapper to the
  * aws-sdk class...
  */
-let wrapSend = (upload) => {
+let wrapSend = (upload, stream) => {
   return new Promise((res, rej) => {
     // TODO: Make this configurable?
+    stream.on('aborted', () => {
+      upload.abort();
+      console.log('upload aborted because of http/https aborted event');
+    });
+    stream.on('error', err => {
+      upload.abort();
+      console.log(err.stack || err);
+    });
     let abortTimer = setTimeout(upload.abort.bind(upload), 1000 * 60 * 60);
```

Comment:
So maybe before we pass the input stream to the `StorageProvider` we do:
```js
inputStream.on('abort', () => inputStream.emit('error', new Error('Request timeout'));
inputStream.on('aborted', () => inputStream.emit('error', new Error('Request aborted'));
inputStream.on('timeout', () => inputStream.abort());
inputStream.setTimeout(1000 * 60 * 60);
```

This way `StorageProvider` only has to worry about events:
 * `data` (handled by `.pipe`)
 * `end` (handled by `.pipe`)
 * `error`